config/slist.c

```diff
@@ -201,6 +201,103 @@ static intptr_t slist_native_get(const struct ConfigSet *cs, void *var,
   return (intptr_t) list;
 }
 
+/**
+ * number_string_plus_equals - Add to a Slist by string - Implements ConfigSetType::string_plus_equals()
+ */
+static int slist_string_plus_equals(const struct ConfigSet *cs, void *var,
+                                     const struct ConfigDef *cdef,
+                                     const char *value, struct Buffer *err)
+{
+  if (!cs || !cdef)
+    return CSR_ERR_CODE; /* LCOV_EXCL_LINE */
+
+  int rc = CSR_SUCCESS;
+
+  /* Store empty strings as NULL */
+  if (value && (value[0] == '\0'))
+    return rc |= CSR_SUC_NO_CHANGE; // return no change
+
+  struct Slist *orig = *(struct Slist **) var;
+  struct Slist *copy = slist_dup(orig);
+
+  if (slist_is_member(copy, value))
+  {
+    slist_free(&copy);
+    return rc |= CSR_SUC_NO_CHANGE; // return no change
+  }
+
+  if (!copy)
+  {
+    mutt_list_insert_tail(&copy->head, mutt_str_strdup(value));
```

Comment:
I'm not sure this is correct

Reply:
Thanks for pointing out. Now I see.

I thought it's necessary when the list is empty to do it other way, because [slist_add_string()](https://neomutt.org/code/mutt_2slist_8c.html#a56a8eca5b8c63f928ae9944661a208bd) has following code at the beginning of the function:
``` 
if (!list)
     return NULL;
```

I tested it wrong, there are two different situations:

1. Make list empty by removing last remaining member:
```
set hidden_tags = test  # let's set it some value
set hidden_tags -= test # remove it, so the list is empty
set hidden_tags += test #set it back
```
2. set list empty at the beginning
```
set hidden_tags = "" # set it empty in the beginning
set hidden_tags += test # segfaults.
```
1. works (hits the else condition branch)
2. segfaults (hits the if condition branch)

slist_add_string() works fine when the slist variable is emptied after removing the last member but not when the variable is set as empty string.

Reply:
this seems to work to me. This way it doesn't take the variables flags from cdef so `set hidden_tags += xyz,abc` inserts one element xyz,abc in both cases 1) and 2)
```
  if (!copy)
    copy = slist_parse(value, 0);
  else
    slist_add_string(copy, value); // add value to list
```

What do you think, @gahr?